extensions/cornerstone-dicom-sr/src/viewports/OHIFCornerstoneSRViewport.tsx

```diff
@@ -0,0 +1,450 @@
+
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  let childrenWithProps = null;
+
+  if (!activeImageDisplaySetData || !referencedDisplaySetMetadata) {
+    return null;
+  }
+
+  if (children && children.length) {
+    childrenWithProps = children.map((child, index) => {
+      return (
+        child &&
+        React.cloneElement(child, {
+          viewportIndex,
+          key: index,
+        })
+      );
+    });
+  }
+
+  const { Modality } = srDisplaySet;
+
+  const {
+    PatientID,
+    PatientName,
+    PatientSex,
+    PatientAge,
+    SliceThickness,
+    ManufacturerModelName,
+    StudyDate,
+    SeriesDescription,
+    SpacingBetweenSlices,
+    SeriesNumber,
+  } = referencedDisplaySetMetadata;
+
+  const label = viewports.length > 1 ? _viewportLabels[viewportIndex] : '';
+
+  // TODO -> disabled double click for now: onDoubleClick={_onDoubleClick}
+  return (
+    <>
+      <ViewportActionBar
+        onDoubleClick={evt => {
+          evt.stopPropagation();
+          evt.preventDefault();
+        }}
+        onPillClick={() => {
+          sendTrackedMeasurementsEvent('RESTORE_PROMPT_HYDRATE_SR', {
+            displaySetInstanceUID: srDisplaySet.displaySetInstanceUID,
+            viewportIndex,
+          });
+        }}
+        onSeriesChange={onMeasurementChange}
+        studyData={{
+          label,
+          useAltStyling: true,
+          isTracked: false,
+          isLocked,
+          isRehydratable: srDisplaySet.isRehydratable,
+          isHydrated,
+          studyDate: formatDate(StudyDate),
+          currentSeries: SeriesNumber,
+          seriesDescription: 'sr viewport',
+          modality: Modality,
+          patientInformation: {
+            patientName: PatientName
+              ? OHIF.utils.formatPN(PatientName.Alphabetic)
+              : '',
+            patientSex: PatientSex || '',
+            patientAge: PatientAge || '',
+            MRN: PatientID || '',
+            thickness: SliceThickness ? `${SliceThickness.toFixed(2)}mm` : '',
+            spacing:
+              SpacingBetweenSlices !== undefined
+                ? `${SpacingBetweenSlices.toFixed(2)}mm`
+                : '',
+            scanner: ManufacturerModelName || '',
+          },
+        }}
+      />
+
+      <div className="relative flex flex-row w-full h-full overflow-hidden">
+        {getCornerstone3DViewport()}
+        <div className="absolute w-full">
+          {viewportDialogState.viewportIndex === viewportIndex && (
+            <Notification
+              message={viewportDialogState.message}
+              type={viewportDialogState.type}
+              actions={viewportDialogState.actions}
+              onSubmit={viewportDialogState.onSubmit}
+              onOutsideClick={viewportDialogState.onOutsideClick}
+            />
+          )}
+        </div>
+        {childrenWithProps}
+      </div>
+    </>
+  );
+}
+
+const _viewportLabels = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I'];
```

Comment:
Hmmmm. This is problematic right? Not for the current _mode_. But this viewport has an implicit limitation that it can only be used in HPs with 9 or less viewports.

Reply:
I know, but it wouldn't just name them; that is what I'm understanding